gui/default/index.html

```diff
@@ -344,40 +344,100 @@ <h3 class="panel-title">
+                        Incorrect user name or password.
+                      </p>
+                      <p ng-if="login.errors.failed" class="text-danger" translate>
+                        Login failed, see Syncthing logs for details.
+                      </p>
+                    </div>
+                  </div>
+                </form>
+              </div>
             </div>
-            <div class="col-md-3 text-right">
-              <button type="submit" id="submit" class="btn btn-default" ng-disabled="login.inProgress" translate>Log In</button>
+          </div>
+
+          <hr class="divider" />
+
+          <div class="panel panel-default">
+            <button class="btn panel-heading" data-toggle="collapse" data-parent="#login-options" data-target="#login-option-webauthn" aria-expanded="{{isDefaultLoginOption('webauthn') ? 'true' : 'false'}}">
+              <h4 class="panel-title">
+                <div class="panel-title-text" translate>WebAuthn (Passkey)</div>
+              </h4>
+            </button>
+            <div id="login-option-webauthn" class="panel-collapse collapse {{isDefaultLoginOption('webauthn') ? 'in' : ''}}">
+              <div class="panel-body">
+                <label class="stay-logged-in">
+                  <input type="checkbox" ng-model="login.stayLoggedIn" ng-disabled="login.inProgress">&nbsp;<span translate>Stay logged in</span>
+                </label>
+
+                <div class="form-group text-right">
+                  <button type="button" class="btn btn-primary" ng-click="authenticateWebauthnFinish()" ng-disabled="login.inProgress || webauthn.errors.noCredentials || webauthn.errors.initFailed || locationDoesNotMatchWebauthnRpId() || isLocationInsecure()" tabindex="3">
+                    <span translate>Log in with WebAuthn</span>
+                  </button>
+                </div>
+
+                <p class="text-info" ng-if="!isLocationInsecure() && !webauthnAvailable()" translate>Your browser does not support WebAuthn.</p>
+                <div ng-if="locationDoesNotMatchWebauthnRpId()">
+                  <p class="text-info" translate>Current domain does not match WebAuthn configuration.</p>
+                  <p>
+                    <a href="{{docsURL('users/webauthn#webauthn-custom-gui-address')}}" translate>Syncthing Documentation: Customizing the GUI address</a>
+                  </p>
+                </div>
+                <p class="text-info" ng-if="isLocationInsecure()" translate>WebAuthn requires HTTPS on non-localhost domains.</p>
+                <p class="text-danger" ng-if="webauthn.errors.notRegistered" translate>This device or security key is not registered. Please try a different one.</p>
+                <p class="text-info" ng-if="webauthn.errors.aborted" translate>Authentication aborted by user.</p>
+                <p class="text-danger" ng-if="webauthn.errors.notAllowed" translate>Authentication failed, please try again.</p>
```

Comment:
If it failed because `notAllowed`, trying again seems like bad advice.  What is the failure really about?

Reply:
[`NotAllowedError` is the catch-all error](https://w3c.github.io/webauthn/#sctn-get-request-exceptions) that's returned when the browser doesn't want to give any more detail - for example, because the user canceled, or because the authenticator they tried to use doesn't have an eligible credential, or (it seems) the request was invoked on a raw IP address host. So no, "try again" probably is the best advice we can give in this case.

Reply:
Alright, it still isn't *good* advice to bang your head against the same wall twice (with unchanged parameters), but I can see how some causes may be temporary, such as timing out on the request.  If we don't give this advice, I guess that's what a user will do anyway.  I certainly would after encountering some mysterious, unspecific error message.

Reply:
Ok, I'll take that as no further action needed then?